src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -204,11 +204,14 @@ private void SendFileInternal(string? fileName, ReadOnlySpan<byte> preBuffer, Re
                 Send(preBuffer);
             }
 
-            // Send the file, if any
-            if (fileHandle != null)
+            using (fileHandle)
```

Comment:
This should be done earlier.  You can delete this and just make the line:
```C#
SafeFileHandle? fileHandle = OpenFileHandle(fileName);
```
instead be:
```C#
using SafeFileHandle? fileHandle = OpenFileHandle(fileName);
```